packages/vitest/src/integrations/chai/poll.ts

```diff
@@ -90,9 +80,24 @@ export function createExpectPoll(expect: ExpectStatic): ExpectStatic['poll'] {
               }
               catch (err) {
                 lastError = err
-                intervalId = setTimeout(check, interval)
+                if (!chai.util.flag(assertion, '_isLastPollAttempt')) {
+                  intervalId = setTimeout(check, interval)
+                }
               }
             }
+            timeoutId = setTimeout(async () => {
```

Comment:
let's not make it async

Reply:
i need to be able to await the `check` call. do you want me to use `Promise.then` instead?
can do it NP, just want to make sure we are aligned

Reply:
Yes, I meant using `promise.then.catch`